tests/test_errors.py

```diff
@@ -1288,6 +1290,94 @@ def finalize(self, cfg: Any) -> None:
+            ),
+            key="baz",
+            full_key="subcfg.baz",
+            child_node=lambda cfg: cfg._get_node("baz"),
+            ref_type=NestedInterpolationToMissing.BazParams,
+        ),
+        id="to_object:structured,throw_on_missing_interpolation",
+    ),
+    param(
+        Expected(
+            create=lambda: OmegaConf.structured(StructuredInterpolationKeyError),
+            op=lambda cfg: OmegaConf.to_object(cfg),
+            exception_type=InterpolationKeyError,
+            key="name",
+            msg=("Interpolation key 'bar' not found"),
+            child_node=lambda cfg: cfg._get_node("name"),
+        ),
+        id="to_object:structured,throw_on_interpolation_key_error",
+    ),
+    # to_container throw_on_missing
+    param(
+        Expected(
+            create=lambda: OmegaConf.create(
+                {"missing_val": "???", "subcfg": {"x": "${missing_val}"}}
+            ).subcfg,
+            op=lambda cfg: OmegaConf.to_container(
+                cfg, resolve=True, throw_on_missing=True
+            ),
+            exception_type=InterpolationToMissingValueError,
+            msg=(
+                "MissingMandatoryValue while resolving interpolation: "
+                "Missing mandatory value: missing_val"
+            ),
+            key="x",
+            full_key="subcfg.x",
+            child_node=lambda cfg: cfg._get_node("x"),
+        ),
+        id="to_container:throw_on_missing_interpolation",
+    ),
+    param(
+        Expected(
+            create=lambda: DictConfig("???"),
+            op=lambda cfg: OmegaConf.to_container(cfg, throw_on_missing=True),
+            exception_type=MissingMandatoryValue,
+            msg="Missing mandatory value",
+        ),
+        id="to_container:throw_on_missing,dict",
+    ),
+    param(
+        Expected(
+            create=lambda: ListConfig("???"),
+            op=lambda cfg: OmegaConf.to_container(cfg, throw_on_missing=True),
+            exception_type=MissingMandatoryValue,
+            msg="Missing mandatory value",
+        ),
+        id="to_container:throw_on_missing,list",
+    ),
+    param(
+        Expected(
+            create=lambda: DictConfig({"a": "???"}),
+            op=lambda cfg: OmegaConf.to_container(cfg, throw_on_missing=True),
+            exception_type=MissingMandatoryValue,
+            msg="Missing mandatory value: a",
+            key="a",
+            child_node=lambda cfg: cfg._get_node("a"),
+        ),
+        id="to_container:throw_on_missing,dict_value",
+    ),
+    param(
+        Expected(
+            create=lambda: ListConfig(["???"]),
+            op=lambda cfg: OmegaConf.to_container(cfg, throw_on_missing=True),
+            exception_type=MissingMandatoryValue,
+            msg="Missing mandatory value: 0",
+            key=0,
+            full_key="[0]",
+            child_node=lambda cfg: cfg._get_node(0),
+        ),
+        id="to_container:throw_on_missing,list_item",
+    ),
```

Comment:
This is adding almost 200 lines to an already huge test file.
We should refactor this file somehow at some point.